api/v1beta1/evaljob_types.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// Represent a job's status
+// +kubebuilder:validation:Enum=New;Scheduled;Running;Complete;Cancelled
+type JobState string
+
+const (
+	// The job is just created
+	NewJobState JobState = "New"
+	// The job is scheduled and waiting for available resources to run it
+	ScheduledJobState JobState = "Scheduled"
+	// The job is running
+	RunningJobState JobState = "Running"
+	// The job is complete
+	CompleteJobState JobState = "Complete"
+	// The job is cancelled
+	CancelledJobState JobState = "Cancelled"
+)
+
+// +kubebuilder:validation:Enum=NoReason;Succeeded;Failed;Cancelled
+type Reason string
+
+const (
+	// Job is still running and no final result yet
+	NoReason Reason = "NoReason"
+	// Job finished successfully
+	SucceedReason Reason = "Succeeded"
+	// Job failed
+	FailedReason Reason = "Failed"
+	// Job is cancelled
+	CancelledReason Reason = "Cancelled"
+)
+
+type Arg struct {
+	Name  string `json:"name"`
+	Value string `json:"value,omitempty"`
+}
+
+// EvalJobSpec defines the desired state of EvalJob
+type EvalJobSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Model name
+	Model string `json:"model"`
+	// Args for the model
+	// +optional
+	ModelArgs []Arg `json:"modelArgs,omitempty"`
+	// Evaluation tasks
+	Tasks []string `json:"tasks"`
+	// Sets the number of few-shot examples to place in context
+	// +optional
+	NumFewShot *int `json:"numFewShot,omitempty"`
+	// Accepts an integer, or a float between 0.0 and 1.0 . If passed, will limit
+	// the number of documents to evaluate to the first X documents (if an integer)
+	// per task or first X% of documents per task
+	// +optional
+	Limit string `json:"limit,omitempty"`
+	// Map to `--gen_kwargs` parameter for the underlying library.
+	// +optional
+	GenArgs []Arg `json:"genArgs,omitempty"`
```

Comment:
I think we may want to name it explicitly `genKwargs` for an exact match or name it `generateArguments` to be more explicit

Reply:
I vote for `generateArgs`